src/index.ts

```diff
@@ -1 +1,77 @@
-console.log('Hello World');
+import {Environment, ErrorData, ErrorReport, ErrorStackTrace, ErrorTrace} from "./models";
+import {generateUUIDv4} from "./utils";
+
+const reportVersion = 1;
+
+let sessionId: string;
+
+const stackTraceLineRegex = /^\s+at\s(\S+)\s\((.+):(\d+):(\d+)\)$/;
+
+export default function InitializeErrorTracker() {
+  sessionId = generateUUIDv4();
+
+  window.addEventListener("error", (error: ErrorEvent) => {
```

Comment:
FYI I recommend forcing an error to pop up after this listener as a "self-check" and report back if it works. Some other applications take over the entire error event system.

Also, allow people to attach their own error reporter to this. Provide a function they can call with an ErrorEvent.

Reply:
Doing a self-check can be harmful if there are other error reporters, as they will not ignore the error.

Reply:
That makes sense, I would like to have a flag I can use at least once to check.

Reply:
You can just throw any error and see if it gets processed and logged or sent over the network